File: src/BaekJoon/DivisorMultiplePrimeNumber/Q9020/Q9020_GoldBach.py
"""
# Author    : Lee In Bok 
# Date      : 2023.12.24(Sun)
# Spend Time: 35m 50s
# Runtime   : 31120 KB
# Memory    : 72 ms
# Algoritm  : Prime Number
"""

import sys
input = sys.stdin.readline

MAX = 10001
arr = [True for _ in range(MAX)]

for num in range(2, MAX):
    if not arr[num]:
      continue

    next = num + num
    while next < MAX:
      arr[next] = False
      next = next + num

T = int(input())

for _ in range(T):
  num = int(input())

  for idx in range(num // 2, 0, -1):
    if not arr[idx]:
      continue

    gap = num - idx

    if arr[gap]:
      print(idx, gap)
      break

File: CodeTest/src/BaekJoon/DivisorMultiplePrimeNumber/Q9020/Q9020_GoldBach.java
/**
* Author    : Lee In Bok 
* Date      : 2023.12.24(Sun)
* Spend Time: 35m 50s
* Runtime   : 208 ms
* Memory    : 17004 KB
* Algoritm  : Prime Number
 */

package BaekJoon.DivisorMultiplePrimeNumber.Q9020;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Q9020_GoldBach {

  public static final int MAX = 10001;
  public static void main(String[] args) throws IOException{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringBuilder sb = new StringBuilder();
    int T = Integer.parseInt(br.readLine());
    boolean[] arr = getPrimeNumbers();

    for(int i = 0; i < T; i++){
      int num = Integer.parseInt(br.readLine());

      for(int j = num/2; j <= num; j++){
        if(arr[j]){
          continue;
        }

        int gap = num - j;

        if(!arr[gap]){
          sb.append(gap + " " + j + "\n");
          break;
        }
      }
    }
    System.out.println(sb.toString());
  }

  public static boolean[] getPrimeNumbers(){
    boolean[] arr = new boolean[MAX];

    for(int num = 2; num < MAX; num++){
      if(arr[num]){
        continue;
      }

      int next = num + num;
      while(next < MAX){
        arr[next] = true;
        next += num;
      }
    }

    return arr;
  }
}